controllers/provider-azure/charts/internal/cloud-provider-config/templates/cloud-provider-worker-config.yaml

```diff
@@ -0,0 +1,38 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cloud-provider-worker-config
+  namespace: {{ .Release.Namespace }}
+data:
+  cloudprovider.conf: |
+    cloud: AZUREPUBLICCLOUD
+    resourceGroup: "{{ .Values.resourceGroup }}"
+    location: "{{ .Values.region }}"
+    vnetName: "{{ .Values.vnetName }}"
+    subnetName: "{{ .Values.subnetName }}"
+    securityGroupName: "{{ .Values.securityGroupName }}"
+    routeTableName: "{{ .Values.routeTableName }}"
+    primaryAvailabilitySetName: "{{ .Values.availabilitySetName }}"
+    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
+    aadClientId: "{{ .Values.aadClientId }}"
+    aadClientSecret: "{{ .Values.aadClientSecret }}"
+    tenantId: "{{ .Values.tenantId }}"
+    subscriptionId: "{{ .Values.subscriptionId }}"
+    {{- end}}
+    cloudProviderBackoff: true
+    cloudProviderBackoffRetries: 6
+    cloudProviderBackoffExponent: 1.5
+    cloudProviderBackoffDuration: 5
+    cloudProviderBackoffJitter: 1.0
+    cloudProviderRateLimit: true
+    cloudProviderRateLimitQPS: 10.0
+    cloudProviderRateLimitBucket: 100
+    cloudProviderRateLimitQPSWrite: 10.0
+    cloudProviderRateLimitBucketWrite: 100
+    {{- if semverCompare ">= 1.14" .Values.kubernetesVersion }}
+    cloudProviderBackoffMode: v2
+    {{- end}}
+    {{- if semverCompare ">= 1.15" .Values.kubernetesVersion }}
```

Comment:
I suggest to bundle this together with the other if condition above:

```yaml
    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
    aadClientId: "{{ .Values.aadClientId }}"
    aadClientSecret: "{{ .Values.aadClientSecret }}"
    tenantId: "{{ .Values.tenantId }}"
    subscriptionId: "{{ .Values.subscriptionId }}"
    {{- else }}
    useInstanceMetadata: true
    {{- end }}
```

Again, the fields for the client id, tenant id, etc. could be moved out into a template and then included here.